packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -0,0 +1,96 @@
+# Table Column Sorting HLD
```

Comment:
Based on the backend discussions now I'm wondering if we should have considerations for a backend controlled sorting behavior?

For example, my data set is 1M elements, only the first 10K are sent to the UI. When I push the sort button in the header I don't want the table to sort the local data, I want to send the request to the server, get the response and put that into the table.

Is that a workflow we should / need to figure out right now?

Reply:
Seems like we would want a cancellable sort event:
- User causes a sort action (i.e. changes sort direction in header for a column)
- Table fires an event that sort is changing
- Application code cancels that event to prevent the sort and fetches data from server with sort configuration
- Application loads data in table and sets the sort configuration (programmatically setting sort does not fire events, only user initiated, so we don't get stuck in a loop)
- Table runs sort on the current data and renders (which is already sorted but should be fast anyway)

Reply:
My initial thoughts are:
1. If all we're talking about is a cancellable event during interactive sort, then I don't think there is anything to be done now. That'll be covered as part of interactive sorting.
2. An interesting question is whether or not we want to have some mode (on a column? on the table?) that says don't actually sort the data, just mark it as sorted because the backend has already put it in the desired order.

I lean towards not complicating the API/code to try to handle (2) without having a solid understanding of the problem we're trying to solve (i.e. is performance an issue? is the sort algorithm the backend uses somehow different than the one the table is using & the rows end up in an unexpected/incorrect order? etc.). Therefore, I think it's fine for now to punt on the issue with the thinking that our 'sort' event will be cancellable. I'll add a note to the top of this HLD that this is something to consider when doing interactive sort.

Reply:
Makes sense to me 👍